pkg/services/ngalert/schedule/schedule.go

```diff
@@ -139,21 +152,162 @@ func (sch *schedule) Unpause() error {
 }
 
 func (sch *schedule) Run(ctx context.Context) error {
-	sch.wg.Add(1)
+	var wg sync.WaitGroup
+	wg.Add(2)
 
 	go func() {
+		defer wg.Done()
 		if err := sch.ruleEvaluationLoop(ctx); err != nil {
 			sch.log.Error("failure while running the rule evaluation loop", "err", err)
 		}
 	}()
 
-	sch.wg.Wait()
+	go func() {
+		defer wg.Done()
+		if err := sch.adminConfigSync(ctx); err != nil {
+			sch.log.Error("failure while running the admin configuration sync", "err", err)
+		}
+	}()
+
+	wg.Wait()
 	return nil
 }
 
-func (sch *schedule) ruleEvaluationLoop(ctx context.Context) error {
-	defer sch.wg.Done()
+// SyncAndApplyConfigFromDatabase looks for the admin configuration in the database and adjusts the sender(s) accordingly.
+func (sch *schedule) SyncAndApplyConfigFromDatabase() error {
+	sch.log.Debug("start of admin configuration sync")
+	q := &models.GetOrgAdminConfiguration{}
+	if err := sch.adminConfigStore.GetAdminConfigurations(q); err != nil {
```

Comment:
Do we still need to pass the query parameter?

Reply:
Good shout, I've seen this pattern used within Grafana of passing a Query struct to attach the results to it but I'm not a fan. I've removed it.